src/core/lwip/pbuf.c

```diff
@@ -162,16 +162,26 @@ u8_t pbuf_header(struct pbuf *p, s32_t header_size_increment)
         return 1;
     }
 
-    /* Check that we aren't going to move off the end of the pbuf */
-    if (header_size_increment < 0 && (-header_size_increment) > (s32_t)p->len) {
-        return 1;
-    }
+    if (header_size_increment >= 0) {
```

Comment:
Not sure this refactoring is good in terms of performance. More code lines to fit in cache.